core/backend/src/IModelIncrementalSchemaLocater.ts

```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module iModels
+ */
+import { ECSqlQueryOptions, ECSqlSchemaLocater, ECSqlSchemaLocaterOptions, SchemaKey, SchemaProps } from "@itwin/ecschema-metadata";
+import { QueryBinder, QueryRowFormat } from "@itwin/core-common";
+import { IModelDb } from "./IModelDb";
+
+/**
+ * A [[ECSqlSchemaLocater]]($ecschema-metadata) implementation that uses the [[IModelDb]] to load schemas incrementally.
+ * @beta
+ */
+export class IModelIncrementalSchemaLocater extends ECSqlSchemaLocater {
+  private readonly _iModel: IModelDb;
+
+  /**
+   * Constructs a new IModelIncrementalSchemaLocater instance.
+   * @param iModel The [[IModelDb]] to query.
+   * @param options Optional [[ECSqlSchemaLocaterOptions]]($ecschema-metadata).
+   */
+  constructor(iModel: IModelDb, options?: ECSqlSchemaLocaterOptions) {
+    super(options ?? { useMultipleQueries: true });
+    this._iModel = iModel;
+  }
+
+  /**
+   * Gets the [[IModelDb]] targeted by this schema loader.
+   */
+  public get iModelDb(): IModelDb {
+    return this._iModel;
+  }
+
+  /**
+   * Gets [[SchemaProps]]($ecschema-metadata) for the given [[SchemaKey]]($ecschema-metadata).
+   * This is the full schema json with all elements that are defined in the schema.
+   * @param schemaKey The key of the schema to be resolved.
+   */
+  public async getSchemaProps(schemaKey: SchemaKey): Promise<SchemaProps | undefined> {
+    // To keep the main thread from being blocked in sync loading cases the resolving
+    // is triggered through a timeout. Even if there is no delay, it improves loading
+    // time by ~3x.
+    return new Promise((resolve, reject) => setTimeout(() => {
```

Comment:
Did you consider `setImmediate` instead of `setTImeout` `0`? Similar performance?

Reply:
I looked at it too, however since it was marked deprecated in the MDN docs, I went with the setTimeout.
https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate

> Deprecated: This feature is no longer recommended. Though some browsers might still support it, it may have already been removed from the relevant web standards, may be in the process of being dropped, or may only be kept for compatibility purposes.

Since that schema locater (or better one of it's implementations is suppose to run in web browsers too, we better not use it, even if it does not seem to be deprecated for node.js.